File: src/views/hang_test.py
#!/usr/bin/env python3

import time
import sys

print("Content-Type: text/html\r")
print("\r")
print("<!DOCTYPE html>")
print("<html><head><title>Hanging CGI Test</title></head>")
print("<body>")
print("<h1>CGI Started...</h1>")
print("<p>This CGI will hang indefinitely to test timeout handling.</p>")

# Flush output to ensure headers are sent
sys.stdout.flush()

# Now hang indefinitely
print("<p>About to hang...</p>")
sys.stdout.flush()

# Infinite loop to simulate hung process
while True:
    time.sleep(1)

File: src/views/hang_test.php
<?php
header("Content-Type: text/html");

echo "<!DOCTYPE html>\n";
echo "<html><head><title>PHP Hanging CGI Test</title></head>\n";
echo "<body>\n";
echo "<h1>PHP CGI Started...</h1>\n";
echo "<p>This PHP CGI will hang indefinitely to test timeout handling.</p>\n";

// Flush output to ensure headers are sent
flush();
ob_flush();

echo "<p>About to hang...</p>\n";
flush();
ob_flush();

// Infinite loop to simulate hung process
while (true) {
    sleep(1);
}
?>
